### PR TITLE
[develop] Add correct logic for 3-digit forecast hours in UPP

### LINKED ISSUE #601: RUN_POST tasks fail for forecasts longer than 99 hours
## Expected behavior
RUN_POST should not have any dependence on forecast length, it should happily run on any model output regardless of valid or forecast time.

## Current behavior
run_post_f### tasks fail if ### > 099

## Machines affected
All

## Steps To Reproduce
1. Create an experiment with a forecast length of 100 hours or more. I used this example on Hera:
```
metadata:
  description: |-
    This test checks the capability of the workflow to retrieve from NOAA
    HPSS grib2-formatted output files generated by the FV3GFS external
    model (FCST_LEN_HRS>=100).
user:
  RUN_ENVIR: community
  MACHINE: hera
  ACCOUNT: fv3lam
workflow:
  CCPP_PHYS_SUITE: FV3_GFS_v16
  PREDEF_GRID_NAME: RRFS_CONUS_25km
  DATE_FIRST_CYCL: '2022060112'
  DATE_LAST_CYCL: '2022060112'
  FCST_LEN_HRS: 108
  PREEXISTING_DIR_METHOD: rename
  COMPILER: intel
  EXPT_SUBDIR: get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2022060112_108h_DT_ATMOS_300
task_get_extrn_ics:
  EXTRN_MDL_NAME_ICS: FV3GFS
  FV3GFS_FILE_FMT_ICS: grib2
task_get_extrn_lbcs:
  EXTRN_MDL_NAME_LBCS: FV3GFS
  LBC_SPEC_INTVL_HRS: 12
  FV3GFS_FILE_FMT_LBCS: grib2
  MEM_GET_EXTRN_LBCS: 4G
task_run_fcst:
  DT_ATMOS: 300
platform:
  BUILD_MOD_FN: build_hera_intel
```
2. Observe that all `run_post_f*` tasks fail after forecast hour 99
```
Checking tests available for monitoring...
Starting experiment get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2022060112_108h running
Starting experiment get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2022060112_108h_DT_ATMOS_300 running
Setup complete; monitoring 2 experiments
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f100, jobid=41820341, in state DEAD (FAILED), ran for 12.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f101, jobid=41820342, in state DEAD (FAILED), ran for 12.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f102, jobid=41820343, in state DEAD (FAILED), ran for 15.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f103, jobid=41820344, in state DEAD (FAILED), ran for 13.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f104, jobid=41820345, in state DEAD (FAILED), ran for 17.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f105, jobid=41820346, in state DEAD (FAILED), ran for 17.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f106, jobid=41820347, in state DEAD (FAILED), ran for 12.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:14 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f108, jobid=41820348, in state DEAD (FAILED), ran for 12.0 seconds, exit status=256, try=2 (of 2)
02/08/23 23:25:24 UTC :: FV3LAM_wflow.xml :: Cycle 202206011200, Task run_post_f107, jobid=41820350, in state DEAD (FAILED), ran for 12.0 seconds, exit status=256, try=2 (of 2)
Experiment get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2022060112_108h_DT_ATMOS_300 is DEAD; will no longer monitor.
```
3. Read the log files to see that this is "intended" behavior: The run_post executable actually succeeds, but because of some out-dated logic, the script simply quits for forecast hours with three digits.
```
FATAL ERROR:
ERROR:
  From script:  ""
  Full path to script:  ""
The ${fhr} variable contains a 3-digit integer whose first digit is not
0.  In this case, it is not clear how to set the variable post_fhr used
in constructing the grib2 file names generated by UPP:
  fhr = "100"
Exiting with nonzero status.
End exregional_run_post.sh at Wed Feb  8 23:25:06 UTC 2023 with error code 1 (time elapsed: 00:00:06)
FATAL ERROR:
ERROR:
  From script:  "JREGIONAL_RUN_POST"
  Full path to script:  "/scratch2/BMC/fv3lam/kavulich/UFS/workdir/python_WE2E_script_round_2/ufs-srweather-app/jobs/JREGIONAL_RUN_POST"
Call to ex-script corresponding to J-job "JREGIONAL_RUN_POST" failed.
Exiting with nonzero status.
End JREGIONAL_RUN_POST at Wed Feb  8 23:25:06 UTC 2023 with error code 1 (time elapsed: 00:00:07)
```

## Detailed Description of Fix (optional)
The fix is quite simple: look for the UPP grib2 file names. After forecast hour 99, the naming convention shifts from `.GrbF99` to `.GrbF100`, shifting from two digits to three, so simply add logic to find those files.

### PR DESCRIPTION
**Pull request by mkavulich**
## DESCRIPTION OF CHANGES: 
Adding correct logic for 3-digit forecast hours in UPP. The filenames coming out of UPP are unfortunately hard-coded in that all forecast hours below 100 will have a 2-digit forecast hour in the filename, and beyond that will switch to a 3-digit forecast hour. The current workflow does not attempt to resolve this difference, resulting in UPP tasks failing after (despite the executable completing successfully). This change adds the correct logic to allow UPP tasks to succeed beyond 99 hours.

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## TESTS CONDUCTED: 
Ran a 108-hour forecast (using the `config.yaml` shown in #601) on Hera and Jet, confirmed that the fix works for 3-digit forecast hours. Ran the comprehensive test suite on Jet and no new failures were observed; all post tasks succeeded.

- [x] hera.intel
- [x] jet.intel
- [x] comprehensive tests (Jet)

## DEPENDENCIES:
None

## DOCUMENTATION:
None

## ISSUE: 
Fixes issue mentioned in #601 

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] My changes do not require updates to the documentation (explain).
  - [x] It's a one-line change
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

### CODE REVIEW

File: scripts/exregional_run_post.sh [hunk 1]
```diff
@@ -240,14 +240,8 @@ if [ ${len_fhr} -eq 2 ]; then
...
     post_fhr="${fhr:1}"
-# What should happen in the "else" case?  Would just setting post_fhr to 
-# fhr work?  Need to test.
   else
-    print_err_msg_exit "\
-The \${fhr} variable contains a 3-digit integer whose first digit is not
-0.  In this case, it is not clear how to set the variable post_fhr used
-in constructing the grib2 file names generated by UPP:
-  fhr = \"$fhr\""
+    post_fhr="${fhr}"
```
Comment (christinaholtNOAA) [inline scripts/exregional_run_post.sh:244]: A comment here on WHY this is necessary might help the next person from going down the rabbit hole you described in personal communications.

**Review by christinaholtNOAA — APPROVED** [review]
LGTM.

**Review by MichaelLueken — APPROVED** [review]
@mkavulich This change looks good to me!

**Comment by venitahagerty** [comment]
Machine: hera
Compiler: intel
Job: WE
Repo location: /scratch1/BMC/zrtrr/rrfs_ci/autoci/pr/1237037101/20230210173511/ufs-srweather-app
Build was Successful
Rocoto jobs started
Long term tracking will be done on 10 experiments
If test failed, please make changes and add the following label back:
ci-hera-intel-WE
Experiment Succeeded on hera: pregen_grid_orog_sfc_climo

**Comment by MichaelLueken** [comment]
The Jenkins tests successfully completed for non-Hera machines and the GHA Hera WE2E test successfully completed.  Moving forward with merging.

**Pull request merged**